components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -26,14 +27,54 @@ const messages = defineMessages({
   },
 });
 
-const StepDetailsCrypto = ({ onChange, data }) => {
+/*
+ * Calculates the approximate value for 1 unit of a given crypto currency in the collective currency.
+ * Uses the Cryptonator API (https://www.cryptonator.com/api/)
+ */
+const getCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+  if (!cryptoCurrency || !collectiveCurrency) {
+    return null;
+  }
+  try {
+    const response = await fetch(`https://api.cryptonator.com/api/ticker/${cryptoCurrency}-${collectiveCurrency}`);
+    const body = await response.json();
+    if (!body.success || body.error !== '') {
+      throw new Error(`Cryptonator Error: ${body.error}`);
+    }
+    return body.ticker.price;
+  } catch (error) {
+    // we don't want the user to see any errors; simply don't show the conversion amount
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return null;
+  }
+};
+
+const StepDetailsCrypto = ({ onChange, data, currency }) => {
   const intl = useIntl();
-  const [currencyType, setCurrencyType] = useState(data.currency);
+  const [cryptoCurrencyType, setCryptoCurrencyType] = useState(data.currency);
   const [amount, setAmount] = useState(data.amount);
+  const [convertedAmount, setConvertedAmount] = useState(data.convertedAmount?.amount);
+  const [cryptoExchangeRate, setCryptoExchangeRate] = useState(null);
   const [touched, setTouched] = useState(false);
   const dispatchChange = (field, value) => {
     onChange({ stepDetails: { ...data, [field]: value }, stepSummary: null });
   };
+
+  const storeCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+    const exchangeRate = await getCryptoExchangeRate(cryptoCurrency.value, collectiveCurrency);
+    setCryptoExchangeRate(exchangeRate);
+  };
+
+  useEffect(() => {
+    storeCryptoExchangeRate(cryptoCurrencyType, currency);
+  }, []);
+
+  useEffect(() => {
+    setConvertedAmount(amount * cryptoExchangeRate);
```

Comment:
Actually, in case there is an error, I'm not comfortable with the idea of multiplying `amount` by `null` and relying on a `0` `convertedAmount` as result. I would like to see something more solid.

Reply:
I've handled this by including a null check on `cryptoExchangeRate` as that will be null when there's an error. Let me know what you think. 👍🏽